rfcs/button-component.md

```diff
@@ -0,0 +1,81 @@
+# Button Component
+
+Rough DOM composition:
+
+```
++-------------------------------------------------------------------+
+| BaseButton (display: flex; justifyContent: space-between)         |
+| +-----------------+-----------------------+---------------------+ |
+| |  startEnhancer  |         label         |     endEnhancer     | |
+| |  display:flex   |                       |     display:flex    | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| +-----------------+-----------------------+---------------------+ |
++-------------------------------------------------------------------+
+```
+
+### Exports
+
+* `Button`
+* `StyledButton`
+* `StyledButtonLabel`
+* `StyledStartEnhancer`
+* `StyledEndEnhancer`
+* `KIND`
+* `SIZE`
+* `SHAPE`
+
+### <Button/> API
+
+* `label: node | function` - Optional
+  A label in the button.
+* `startEnhancer: node | function` - Optional
+  A helper rendered at the start of the button.
+* `endEnhancer: node | function` - Optional
+  A helper rendered at the end of the button.
+* `disabled: boolean` - Optional
+  Defines if the button disabled.
+
+### Presentational components props API
+
+These properties are passed to every presentational (styled) component that is exported:
+
+* `$size: 'default' | 'compact'`
+* `$shape: 'round' | 'square'`
+* `$kind: 'primary' | 'secondary' | 'tertiary' | 'minimal'`
+* `$theme: theme`
+
+### Usage
+
+Basic usage:
+
+```javascript
+import * as React from 'react';
+import {
+  Button, 
+  StyledStartEnhancer, 
```

Comment:
if we call it basic usage, should it be really basic, without enchancers, etc? it looks a bit scary if we reference it as basic, imho. how about adding a really basic one, and this?

Reply:
Good point! Thanks.